docs/user-guide/comparison.rst

```diff
@@ -0,0 +1,70 @@
+.. _usage-comparison:
+
+Comparison with Boost.Histogram
```

Comment:
We might want to just link to this page, remove it, or at least prefix it with a comparison to boost-histogram. Users will not be as interested in Boost.Histogram here as they would be in a library called boost-histogram. A page hist vs. boost-histogram is almost what we have in the readme.

Reply:
Alright! Should we show both the comparisons on the same page or different ones?

Reply:
I'd not bother with repeating the comparisons in Hist, users can go to boost-histogram for comparisons with Boost.Histogram. I'd just not have this page at all, I think.

Reply:
Okay, that makes sense!